variables.tf

```diff
@@ -81,3 +81,14 @@ variable "finding_publishing_frequency" {
   type        = string
   default     = null
 }
+
+variable "s3_protection_enabled" {
+  description = <<-DOC
+  Flag to indicate whether S3 protection will be turned on in GuardDuty.
+
+  For more information, see:
+  https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/guardduty_detector
+  DOC
+  type        = bool
+  default     = false
```

Comment:
From the [docs](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/guardduty_detector#s3-logs), this defaults to true, so we should probably keep that behavior for the module.

Reply:
Actually, on further inspection it defaults to true if you specify the data sources block, so your initial implementation is more the expected default behavior. I'll resolve this comment and approve.